kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/discovery/informer/IndexerComposite.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.kubernetes.discovery.informer;
+
+import io.kubernetes.client.common.KubernetesObject;
+import io.kubernetes.client.informer.cache.Indexer;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import org.apache.commons.collections4.map.HashedMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * The {@link io.kubernetes.client.informer.cache.Indexer} composite for the given {@code ApiType} that provides
+ * access to the {@link io.kubernetes.client.informer.cache.Store} resources.
+ *
+ * @param <ApiType> api type of the cache
+ * @author Pavol Gressa
+ * @since 3.2
+ */
+@Internal
+public class IndexerComposite<ApiType extends KubernetesObject> {
```

Comment:
This is just an abstraction over the Indexers. The thing is that all informers has an eventually consistent list of watcher resources. Also the informers are namespace scoped. So in case of searching for a resource with name and namespace, we need to find the informer for given namespace and lookup for the resource by its name. And that's this composite about.